src/main/scala/chisel3/util/Counter.scala

```diff
@@ -62,13 +65,8 @@ object Counter
   @chiselName
   def apply(cond: Bool, n: Int): (UInt, Bool) = {
     val c = new Counter(n)
-    // Note this use of var is generally frowned upon! Don't use this as an example.
-    // This is done because we wanted the hardware generated by c.inc() to be wrapped
-    // in a when statement, but still needed to refer to the final value returned by
-    // c.inc() so we could return cond && wrap. Unless you really, really know what
-    // you are doing, IGNORE THIS CODE!!!!!
-    var wrap: Bool = null
```

Comment:
mea culpa, mea culpa, mea maxima culpa

Reply:
To be fair, there are 2 other issues:
* https://github.com/freechipsproject/chisel3/issues/1409
* https://github.com/freechipsproject/firrtl/issues/1505